core/isolate.rs

```diff
@@ -244,8 +402,340 @@ impl Isolate {
       op_registry: Arc::new(OpRegistry::new()),
       waker: AtomicWaker::new(),
       error_handler: None,
+    };
+
+    // TODO: do it later
+    // unsafe { isolate.set_data(0, self_ptr as *mut c_void) };
+    eprintln!("before");
+    {
+      let core_isolate_ptr: *mut Self = &mut core_isolate;
+      unsafe { isolate.set_data(0, core_isolate_ptr as *mut c_void) };
+      core_isolate.isolate_ = Some(isolate);
+    }
+    eprintln!("after");
+
+    core_isolate
```

Comment:
@ry this is probably bad, I've got tests crashing when retrieving isolate using `get_data()`